hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/JetSqlValidator.java

```diff
@@ -59,27 +62,61 @@ public SqlNode validate(SqlNode topNode) {
         SqlNode validated = super.validate(topNode);
 
         if (validated instanceof SqlInsert) {
-            if (!isCreateJob && isInsertFromStreamingSource((SqlInsert) validated)) {
+            SqlInsert insert = ((SqlInsert) validated);
+            if (!isCreateJob && containsStreamingSource(insert.getSource())) {
                 throw newValidationError(topNode, RESOURCE.mustUseCreateJob());
             }
         }
         return validated;
     }
 
+    @Override
+    protected void validateGroupClause(SqlSelect select) {
+        super.validateGroupClause(select);
+
+        if (containsGroupingOrAggregation(select) && containsStreamingSource(select)) {
```

Comment:
This condition will wrongly reject this query:
```sql
SELECT * FROM my_stream
WHERE a=(SELECT COUNT(*) FROM my_table)
```
We don't support joins yet though. I think we should not check for aggregations on a deeper level. On every SELECT we should check if that SELECT groups or aggregates, and if it does, check that there's no streaming input anywhere.

In the above example, the SELECT on line 1 doesn't aggregate or group, so no check for streaming sources. The SELECT on line 2 does aggregate, so check its inputs: its input is a table, so it's ok.